compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/converters/FileSystemSourceInput.java

```diff
@@ -15,15 +16,23 @@
 public class FileSystemSourceInput implements CompilerInput {
 
     private final Path path;
+    private Path packageRoot;
 
     public FileSystemSourceInput(Path path) {
         this.path = path;
     }
 
+    public FileSystemSourceInput(Path filePath, Path packageRoot) {
+        this.path = filePath;
+        this.packageRoot = packageRoot;
+    }
+
     @Override
     public String getEntryName() {
         Path fileName = path.getFileName();
-        return fileName == null ? path.toString() : fileName.toString();
+        return packageRoot != null ?
+                new File(packageRoot.toString()).toURI().relativize(new File(path.toString()).toURI()).getPath() :
+                (fileName != null ? fileName.toString() : path.toString());
```

Comment:
Too many things cluttered in to a single statement. Shall we use a simple if-else?

Reply:
Also add some comments to explain what you are doing here.

Reply:
Ack; Addressed by commit 0c9d4fd.